src/server/scripts/Shadowlands/Sanctum of Domination/boss_sylvanas_windrunner.cpp

```diff
@@ -0,0 +1,3702 @@
+
+            case ACTION_DESECRATING_SHOT_LAUNCH:
+            {
+                if (Creature* shadowCopy = ObjectAccessor::GetCreature(*me, _shadowCopyGUID[0]))
+                {
+                    shadowCopy->NearTeleportTo(me->GetPosition(), false);
+
+                    me->SendPlaySpellVisualKit(VISUAL_KIT_SYLVANAS_SHADOW_DAGGER, 0, 0);
+                    //me->SendPlaySpellVisualKit(VISUAL_KIT_SYLVANAS_UNSHEATHE_DAGGERS_SPIN, 0, 0);
+
+                    me->SendPlaySpellVisualKit(VISUAL_KIT_SYLVANAS_SHADOW_DAGGER, 0, 0);
+                    DoCastSelf(SPELL_RIVE_DISAPPEAR, true);
+                    DoAction(ACTION_WINDRUNNER_MODEL);
+
+                    shadowCopy->NearTeleportTo(me->GetNearPosition(5.0f, float(M_PI)), false);
+
+                    scheduler.Schedule(50ms, [this, shadowCopy](TaskContext context)
+                    {
+                         shadowCopy->NearTeleportTo(shadowCopy->GetPositionX(), shadowCopy->GetPositionY(), shadowCopy->GetPositionZ() + 5.0f, shadowCopy->GetOrientation(), false);
+                    });
+
+                    scheduler.Schedule(150ms, [this, shadowCopy](TaskContext context)
+                    {
+                         if (!_sayDesecratingOnCD)
+                         {
+                             _sayDesecratingOnCD = true;
+
+                             Talk(SAY_DESECRATING_SHOT);
+                         }
+
+                         DoAction(ACTION_WINDRUNNER_MODEL);
+
+                         me->SetNameplateAttachToGUID(_shadowCopyGUID[0]);
+
+                         shadowCopy->CastSpell(shadowCopy, SPELL_DESECRATING_SHOT_JUMP_FRONT, true);
+                    });
+
+                    scheduler.Schedule(450ms, [this, shadowCopy](TaskContext context)
+                    {
+                         shadowCopy->CastSpell(me->GetPosition(), SPELL_WINDRUNNER_MOVE, true);
+                    });
+
+                    scheduler.Schedule(650ms, [this, shadowCopy](TaskContext context)
+                    {
+                         me->SetNameplateAttachToGUID(ObjectGuid::Empty);
+
+                         me->SendPlaySpellVisualKit(VISUAL_KIT_SYLVANAS_SHADOW_DAGGER, 0, 0);
+                    });
+                }
+
+                break;
+            }
+
+            case ACTION_PREPARE_PHASE_THREE:
+            {
+                DoAction(ACTION_PAUSE_AI_FOR_EVENT);
+
+                DoCastSelf(SPELL_BANSHEE_SHROUD, true);
+
+                if (Creature* jaina = instance->GetCreature(DATA_JAINA_PROUDMOORE_PINNACLE))
+                {
+                    if (jaina->IsAIEnabled())
+                        jaina->AI()->DoAction(ACTION_OPEN_PORTAL_TO_PHASE_THREE);
+                }
+
+                break;
+            }
+
+            case ACTION_INITIATE_PHASE_THREE:
+            {
+                me->NearTeleportTo(SylvanasPhase3Pre, false);
+
+                if (Creature* shadowCopy = ObjectAccessor::GetCreature(*me, _shadowCopyGUID[0]))
```

Comment:
Use a for loop for each copy if you will do the same thing.

Reply:
fixed on next commit.